src/main/java/hlf/java/rest/client/service/impl/TransactionFulfillmentImpl.java

```diff
@@ -47,10 +53,121 @@
 @Service
 public class TransactionFulfillmentImpl implements TransactionFulfillment {
 
+  private static final long DEFAULT_ORDERER_TIMEOUT = 60;
+  private static final TimeUnit DEFAULT_ORDERER_TIMEOUT_UNIT = TimeUnit.SECONDS;
+
+  private static final long DEFAULT_COMMIT_TIMEOUT = 5;
+  private static final TimeUnit DEFAULT_COMMIT_TIMEOUT_UNIT = TimeUnit.MINUTES;
+
   @Autowired private FabricProperties fabricProperties;
 
   @Autowired private Gateway gateway;
 
+  @Autowired private HFClientWrapper hfClientWrapper;
+
+  @Override
+  public ResponseEntity<ClientResponseModel> initSmartContract(
+      String networkName,
+      String contractName,
+      String functionName,
+      Optional<List<String>> peerNames,
+      String... transactionParams) {
+    log.info("initialize the smart contract, this is done once every install");
+    // get the network information to send the request
+    Network network = gateway.getNetwork(networkName);
+
+    // set the init flag
+    Channel channel = network.getChannel();
+
+    // Generate the transaction proposal, set the init flag
+    TransactionProposalRequest transactionProposalRequest =
+        hfClientWrapper.getHfClient().newTransactionProposalRequest();
+    transactionProposalRequest.setInit(true);
+    transactionProposalRequest.setChaincodeName(contractName);
+    transactionProposalRequest.setFcn(functionName);
+    transactionProposalRequest.setTransactionContext(channel.newTransactionContext());
+    transactionProposalRequest.setArgs(transactionParams);
+
```

Comment:
Q: can init transaction have the transient data?

Reply:
I thought about it. What would be use if the data shall be in the private ledger that is not shared with anybody else. It probably will not be used in the smart contract decision making. What do you say?

Reply:
Well, this goes towards the usecase and I agree with you but what do you say if we can add that capability since this may or maynot be need of usecase.